PhysicsTools/PatAlgos/python/producersLayer1/lowPtElectronProducer_cff.py

```diff
@@ -71,10 +69,22 @@
 
 # Schedule rekeying of seed BDT ValueMaps by reco::GsfElectron for run2_miniAOD_UL and bParking
 from Configuration.ProcessModifiers.run2_miniAOD_UL_cff import run2_miniAOD_UL
+from Configuration.Eras.Modifier_run2_miniAOD_devel_cff import run2_miniAOD_devel
 from Configuration.Eras.Modifier_bParking_cff import bParking
+
 from RecoEgamma.EgammaElectronProducers.lowPtGsfElectronSeedValueMaps_cff import rekeyLowPtGsfElectronSeedValueMaps
 from RecoEgamma.EgammaElectronProducers.lowPtGsfElectronID_cff import lowPtGsfElectronID
+from RecoEgamma.EgammaElectronProducers.lowPtGsfElectrons_cff import lowPtGsfElectrons
+
 _makePatLowPtElectronsTask = makePatLowPtElectronsTask.copy()
+_makePatLowPtElectronsTask2 = makePatLowPtElectronsTask.copy()
+
 _makePatLowPtElectronsTask.add(rekeyLowPtGsfElectronSeedValueMaps)
 _makePatLowPtElectronsTask.add(lowPtGsfElectronID)
-(bParking | run2_miniAOD_UL).toReplaceWith(makePatLowPtElectronsTask,_makePatLowPtElectronsTask)
+(~bParking & run2_miniAOD_UL).toReplaceWith(makePatLowPtElectronsTask,_makePatLowPtElectronsTask) # no-change rule
+
+_makePatLowPtElectronsTask2.add(rekeyLowPtGsfElectronSeedValueMaps)
+_makePatLowPtElectronsTask2.add(lowPtGsfElectronID)
+_makePatLowPtElectronsTask2.add(lowPtGsfElectrons)
+(~bParking & run2_miniAOD_devel).toReplaceWith(makePatLowPtElectronsTask,_makePatLowPtElectronsTask2)
+(bParking & run2_miniAOD_UL).toReplaceWith(makePatLowPtElectronsTask,_makePatLowPtElectronsTask2)
```

Comment:
```suggestion
run2_miniAOD_UL.toReplaceWith(makePatLowPtElectronsTask,_makePatLowPtElectronsTask)

((bParking & run2_miniAOD_UL) | (~bParking & run2_miniAOD_devel)).toModify(
    makePatLowPtElectronsTask, func = lambda t: t.add(lowPtGsfElectrons))
```
- `_devel` implies it's added on top of `_UL`, which simplifies the first toReplaceWith
- the addition of `lowPtGsfElectrons` can be done directly, which shortens its addition in the second modification call

Just to make sure I understand though: if the change is done consistently with the reco updates in `bParking` and reco/AOD is done in the same release as this PR, technically we do not need to rerun the ID and regression if `bParking` is enabled; but we still rerun ID and regression here to decouple from the possible last minute decision to run reco/AOD `bParking` in an older release. Is this correct?
Please confirm.

Reply:
Again, I agree appending rather than copy/paste is perhaps more elegant but I couldn't get the desired behaviour. (Likely I had a bug.)

Further, while I agree copy/paste is less elegant, I just find the current code much easier to read than your suggestion. Personally, I'd prefer to leave the code as is if possible ...? 

"Just to make sure I understand though: if the change is done consistently with the reco updates in bParking and reco/AOD is done in the same release as this PR, technically we do not need to rerun the ID and regression if bParking is enabled; but we still rerun ID and regression here to decouple from the possible last minute decision to run reco/AOD bParking in an older release. Is this correct?
Please confirm."

Correct.

Reply:
> Further, while I agree copy/paste is less elegant, I just find the current code much easier to read than your suggestion. Personally, I'd prefer to leave the code as is if possible ...?

it looks more clear to me the way I proposed: UL is clearly unchanged and bParking or devel simply add on top of it. Otherwise it's harder to read and see that bParking is simply extends extend of doing something else from scratch

Reply:
Changed as suggested.